staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -244,7 +245,12 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		isCreater = true
 	}
 
-	var versionedList interface{}
+	var (
+		versionedList             interface{}
+		versionedListOptions      runtime.Object
+		versionedListExtraOptions runtime.Object
+		listSubpath               bool
```

Comment:
What is `versionedListExtraOptions` for? Why is `versionedListOptions` insufficient?  Should we bundle the various vars together into a struct?

Reply:
`versionedListOptions` and `versionedListExtraOptions` are two different kind of objects. `versionedListOptions` is of `ListOptions` and contains the standard `LIST` operation options. `versionedListExtraOptions` is of user provided type and is meant to contain user-defined options in addition to the standard options.